rdwatch/core/views/animation.py

```diff
@@ -0,0 +1,211 @@
+
+@router.delete(
+    '/site/{id}/',
+    response={
+        403: DeleteErrorSchema,
+        200: DeleteSuccessSchema,
+    },
+)
+def delete_site_download(
+    request: HttpRequest,
+    id: UUID4,
+):
+    # Fetch the export object
+    export = get_object_or_404(AnimationSiteExport, celery_id=id, user=request.user)
+
+    # Perform the delete operation
+    export.delete()
+    return 200, {'success': 'Site Animation Export deleted successfully.'}
+
+
+@router.delete(
+    '/modelrun/{id}/',
+    response={
+        403: DeleteErrorSchema,
+        200: DeleteSuccessSchema,
+    },
+)
+def delete_modelrun_download(
+    request: HttpRequest,
+    id: UUID4,
+):
+    # Fetch the export object
+    export = get_object_or_404(AnimationModelRunExport, celery_id=id, user=request.user)
+
+    # Perform the delete operation
+    export.delete()
+    return 200, {'success': 'ModelRun Animation Export deleted successfully.'}
+
+
+@router.post('/site/{id}/debug/')
+def generate_animation_debug(
+    request: HttpRequest,
+    id: UUID4,
+    params: GenerateAnimationSchema,  # noqa: B008
+):
+    # Fetch the SiteEvaluation instance
+    datapath, name = create_animation(
+        site_evaluation_id=id,
+        settings=params.dict(),
+    )
+    if datapath:
+        with open(datapath, 'rb') as f:
+            response = HttpResponse(f.read(), content_type='application/octet-stream')
+            response['Content-Disposition'] = f'attachment; filename={name}'
+            return response
+    return 500, 'Unable to export data'
+
+
+@router.post('/modelrun/{id}/')
+def generate_modelrun_animation(
+    request: HttpRequest,
+    id: UUID4,
+    params: GenerateAnimationSchema,  # noqa: B008
+):
+    # Fetch the SiteEvaluation instance
+    task_id = create_modelrun_animation_export.delay(
+        modelrun_id=id, settings=params.dict(), userId=request.user.pk
+    )
+    return task_id.id
+
+
+@router.get('/download/site/{task_id}/')
+def get_downloaded_site_animation(request: HttpRequest, task_id: UUID4):
+    animation_export = get_object_or_404(
+        AnimationSiteExport, celery_id=task_id, user=request.user
+    )
+    name = animation_export.name
+    content_type = 'video/mp4'
+    if name.endswith('.gif'):
+        content_type = 'image/gif'
+    response = HttpResponse(
+        animation_export.export_file.file, content_type=content_type
+    )
+    response['Content-Disposition'] = f'attachment; filename="{name}"'
+    return response
+
+
+@router.get('/download/modelrun/{task_id}/')
+def get_downloaded_modelrun_animation(request: HttpRequest, task_id: UUID4):
+    animation_export = get_object_or_404(
+        AnimationModelRunExport, celery_id=task_id, user=request.user
+    )
+    name = animation_export.configuration.title
+    response = HttpResponse(
+        animation_export.export_file.file, content_type='application/zip'
+    )
+    response['Content-Disposition'] = f'attachment; filename="{name}.zip"'
+    return response
```

Comment:
I think we should avoid buffering the entire file into memory in a server request thread. Instead, the usual pattern is to generate a presigned url and redirect to s3/minio.

```suggestion
from django.shortcuts import redirect

@router.get('/download/site/{task_id}/')
def get_downloaded_site_animation(request: HttpRequest, task_id: UUID4):
    return redirect(get_object_or_404(
        AnimationSiteExport, celery_id=task_id, user=request.user
    ), permanent=False)


@router.get('/download/modelrun/{task_id}/')
def get_downloaded_modelrun_animation(request: HttpRequest, task_id: UUID4):
    return redirect(get_object_or_404(
        AnimationModelRunExport, celery_id=task_id, user=request.user
    ).export_file.url, permanent=False)
```

@BryonLewis is it necessary to set the Content-Disposition headers? If so, I think we can do that but it makes things a bit more complicated.

Reply:
I'd have to double check if without attachment it properly downloads.

Reply:
Looks like it won't handle it properly.

Reply:
I think we should still fix this, but I'm fine with merging this as-is to avoid bloating this PR more as the solution might not be straightforward.

@zachmullen this is what I was referring to when we were discussing generated presigned urls w/ a Content-Disposition header set. I'm curious what your thoughts are on leaving this as-is vs. generating a presigned url with correct headers (at the cost of more code complexity)? My instinct is that the in-memory approach could cause unforeseen issues with memory usage/response times, but it's also possible I'm overthinking this.

Reply:
To avoid the issue with memory, use a [`django.http.response.FileResponse`](FileResponse) rather than a normal `HttpResponse`. That will avoid the issue of loading the whole file into memory. The downside of this approach is that it will require Django connections to be occupied serving binary download of potentially large files, which is something we typically want to avoid, particularly under WSGI as opposed to ASGI. From the Django docs:

> There are performance considerations when doing this, though. Django, under WSGI, is designed for short-lived requests. Streaming responses will tie a worker process for the entire duration of the response. This may result in poor performance.

Maybe the best thing to do is have a sync-up to understand the requirements around your content-disposition headers in the first place.

Reply:
There may be a way I can directly reference the presigned URL.  I didn't have the time to properly debug it earlier today so I tried the suggested code and swapped back after it didn't work.

Reply:
I modified the front-end so I can send it just the pre-signed URL for downloading and a filename to use.
I just need to do a bit more testing on it tomorrow morning,

Reply:
Perfect, looks like this approach works :+1: